packages/react-components/react-swatch-picker-preview/src/utils/contrastUtils.ts

```diff
@@ -0,0 +1,108 @@
+export type Rgb = [number, number, number] | null;
+
+export function convertColorToRgb(color: string): Rgb {
+  if (color.includes('rgb') || color.includes('rgba')) {
+    const rgb = color.replace(/[^\d,]/g, '').split(',');
+    return [parseInt(rgb[0], 10), parseInt(rgb[1], 10), parseInt(rgb[2], 10)];
+  }
+  if (color.includes('#')) {
+    return hexToRgb(color);
+  }
+  return null;
+}
```

Comment:
I would say that it's _too naive_ approach: there are so many values to define colors (https://developer.mozilla.org/en-US/docs/Web/CSS/color_value) including the CSS variables (somebody can just pass `--my-color`), it looks error-prone...

There is `color-contrast()` in CSS (https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-contrast) which is sadly supported nowhere. I suggest to check following:
- https://github.com/w3c/csswg-drafts/issues/7553
- https://observablehq.com/@argyleink/contrast-color (has a good example on bottom of the page)

Reply:
Unfortunately there are not supported yet. For now we'll leave for partners to calculate contrast ratio and for disabled icon I added shadow